### PR TITLE
[feature] Allow shared subnets to have non shared child subnets #90

### LINKED ISSUE #90: [feature] Allow shared subnets to have non shared child subnets
**Steps to replicate**

1. Create a shared subnet `10.0.0.0/16`
2. Try to create a child subnet of `10.0.0.0/16` under the `default` organization. 

You will see the following error: 
```
Please ensure that the organization of this subnet and the organization of the related subnet match.
```
**Expected behaviour**

Subnet should get created without any error.

### PR DESCRIPTION
**Pull request by pandafy**
Closes #90

### CODE REVIEW

File: openwisp_ipam/base/models.py [hunk 1]
```diff
@@ -82,20 +84,35 @@ def _validate_multitenant_master_subnet(self):
...
             )
 
     def _validate_overlapping_subnets(self):
-        qs = self._meta.model.objects.filter(organization=self.organization).only(
-            'subnet'
-        )
+        organization_query = Q(organization_id=self.organization_id)
+        error_message = _('Subnet overlaps with {0}.')
+        if self.master_subnet:
+            if self.master_subnet.organization_id is None:
```
Comment (nemesisdesign) [inline openwisp_ipam/base/models.py:105]: this works if there's only 1 level, but may not work if the master subnet does have an org set but has itself a relation to another master subnet which is shared, that's why in one of the validation checks below we iterate over all the master subnets

**Review by nemesisdesign — COMMENTED** [review]
can you rebase on the current master and see my comment below?

**Review by nemesisdesign — CHANGES_REQUESTED** [review]
Tested and works. I am not sure it will work with nested master subnets, if you can check it would be great, if it doesn't work we can create an issue to deal with that later on.

Please fix the conflict and solve the issue highlighted in the comment below.

**Comment by pandafy** [comment]
@nemesisdesign did you mean to test this? 
image

Coverage seem to go down for unrelated reason.

**Comment by nemesisdesign** [comment]
> @nemesisdesign did you mean to test this?
> image

Yes, does it work?
Is there a test for it?
 
> Coverage seem to go down for unrelated reason.

Ok, let's ignore it now, I created an issue for this: https://github.com/openwisp/openwisp-ipam/issues/96

**Comment by pandafy** [comment]
> Is there a test for it? 

I have added a test for it.

**Review by nemesisdesign — APPROVED** [review]
Thanks!

**Pull request merged**